sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -1385,8 +1389,37 @@ case class DropIndex(
 case class TableSpec(
     properties: Map[String, String],
     provider: Option[String],
-    options: Map[String, String],
+    // Note that the map value can be null if the input option only provided a key.
+    optionsList: Map[String, Expression],
     location: Option[String],
     comment: Option[String],
     serde: Option[SerdeInfo],
-    external: Boolean)
+    external: Boolean) extends TreePatternBits {
+  /** This is a convenience method to obtain the options list as a map of strings to strings. */
+  def options: Map[String, String] = optionsList.map { case (key, value) =>
+    val newValue = if (value == null) {
+      null
+    } else value match {
+      case Literal(litVal: String, _) => litVal
+      case Literal(litVal: UTF8String, _) => litVal.toString
+      case _ => value.sql
```

Comment:
How can a non-literal expression convert back to itself with `toExpressionOptions`?

Reply:
I realize that in this case, the constant-folding step was not able to convert the option value to a literal, so it was non-constant; we return an appropriate error message in this case now.